SetData/SetData.cs

```diff
@@ -1033,18 +1033,18 @@ internal static void SetPieceRecipeData(PieceData data, ObjectDB Instant, bool c
                                         foreach (Renderer renderitem in renderfinder)
                                         {
                                             if (renderitem.receiveShadows && materialstr[0] != "none")
-                                                renderitem.material = mat;
```

Comment:
Accessing 'material' will create a new instance of a material. So instead we're updating sharedMaterial.